src/AdvancedMH.jl

```diff
@@ -51,29 +51,39 @@ logdensity(model::DensityModel, t::Transition) = t.lp
 
 # A basic chains constructor that works with the Transition struct we defined.
 function AbstractMCMC.bundle_samples(
-    ts::Vector{<:Transition},
+    ts::Vector{Transition{T, L}},
     model::DensityModel,
     sampler::MHSampler,
     state,
     chain_type::Type{Vector{NamedTuple}};
     param_names=missing,
     kwargs...
-)
-    # Check if we received any parameter names.
-    if ismissing(param_names)
-        param_names = ["param_$i" for i in 1:length(keys(ts[1].params))]
+) where {T, L}
+    # If the element type of ts is NamedTuples, just use those names:
+    if T <: NamedTuple
```

Comment:
Could we use dispatch here instead of hardcoding a check? That would make it more flexible and easier extendable.

Reply:
Yeah, works for me.